libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1489,6 +1489,23 @@ bool TypeChecker::visit(FunctionCall const& _functionCall)
 		}
 	}
 
+	if (auto memberAccess = dynamic_cast<MemberAccess const*>(&_functionCall.expression()))
```

Comment:
Wouldn't it be better to check already at `MemberAccess`?
As it currently is, `a.transfer` would not generate an error, only `a.transfer(x)` would.